deploy/lib/buildAndPushContainers.js

```diff
@@ -5,15 +5,37 @@ const tar = require('tar-fs');
 
 const docker = new Docker();
 
-const promisifyStream = (stream, verbose) => new Promise((resolve, reject) => {
-  stream.on('data', (data) => {
-    if (verbose) {
-      console.log(data.toString().replace('\n', ''));
-    }
+function extractStreamContents(stream, verbose) {
+  return new Promise((resolve, reject) => {
+    const streamContent = [];
+
+    stream.on('data', (data) => {
+      const streamData = data.toString().replace('\n', '')
+      streamContent.push(streamData);
+
+      if (verbose) {
+        console.log(streamData);
+      }
+    });
+
+    stream.on('end', () => {
+      resolve(streamContent);
+    });
+    stream.on('error', reject);
   });
-  stream.on('end', resolve);
-  stream.on('error', reject);
-});
+}
+
+function findErrorInBuildOutput(buildOutput) {
+  for (let i = 0; i < buildOutput.length; i++) {
+    if (buildOutput[i].startsWith('{"errorDetail":{')) {
+      const errorDetail = JSON.parse(buildOutput[i])['errorDetail'];
```

Comment:
JSON.parse [can throw exceptions](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse#exceptions)

can you surround it with try/catch ?

```javascript
let errorDetail;
try {
errorDetail = JSON.parse(buildOutput[i])['errorDetail'];
}
catch(err) {
return "" // (or something undefinied ?)
}
try {

```

Reply:
I've added a try/catch :)